test/test_uop_graph.py

```diff
@@ -119,16 +119,25 @@ def test_const_cast(self):
     self.assertEqual(out.op, UOps.CONST)
     self.assertEqual(out.arg, 0)
 
+  def test_const_vectorize(self):
+    bf = UOp(UOps.CONST, dtypes.bool, arg=False)
+    out = UOp(UOps.VECTORIZE, dtypes.int, (bf,))
```

Comment:
yea this test is actually wrong, you can't vectorize something to another dtype. you first need to cast it.
eg: bool -> CAST int -> vectorize int4
also how doesn't the uops type_verify catch scalar dtypes.int on vectorize?

Reply:
Ah, yeah, I've just been thinking about vectorize as a variant of cast, but what you're saying makes sense. I'll take another stab at it and think a little harder about what vectorize should be doing.

Reply:
> also how doesn't the uops type_verify catch scalar dtypes.int on vectorize?
I think this is because the type_verify happens after the fold, so the VECTORIZE is removed before the check happens. But if I take out the rule that I added, it does indeed hit the assert.

Reply:
ahh right - It's ok if it's just fixed for this test.

Reply:
> good moment to also add the dtype assert.

I added an assert, lmk what you think: https://github.com/tinygrad/tinygrad/pull/5289/files#diff-91ca5e2e75ef3ea1982c8ca6cc175ee88f20efa0d8e4b96f305b970dc6df71e7R102

Trying to assert "dtype should be a vec()'d version of src.dtype"

Reply:
Oh and I think I made my test sane. I constructed it based on what I was seeing in the prior process_replay failures as opposed to doing nonsensical things with the dtypes lol.